frontend/src/pages/forms/Form.jsx

```diff
@@ -29,11 +51,70 @@ const Form = () => {
   const [loading, setLoading] = useState(false);
   const [alertMessage, setAlertMessage] = useState('');
   const [successful, setSuccessful] = useState(false);
+  const [tokenPrice, setTokenPrice] = useState(0);
+  const [healthFactor, setHealthFactor] = useState(0);
+  
   useLockBodyScroll(successful);
   const [isClosePositionOpen, setClosePositionOpen] = useState(false);
   const connectWalletMutation = useConnectWallet(setWalletId);
   const { data: positionData, refetch: refetchPosition } = useCheckPosition();
 
+  // Fetch token price from CoinGecko
+  useEffect(() => {
+    const fetchTokenPrice = async () => {
+      try {
+        const tokenId = TOKEN_CONFIG[selectedToken].id;
+        const response = await fetch(
+          `https://api.coingecko.com/api/v3/simple/price?ids=${tokenId}&vs_currencies=usd`
+        );
+        const data = await response.json();
+        setTokenPrice(data[tokenId].usd);
+      } catch (error) {
+        console.error('Error fetching price:', error);
+        setTokenPrice(0);
+      }
+    };
+
+    if (selectedToken) {
+      fetchTokenPrice();
+    }
+  }, [selectedToken]);
```

Comment:
fetching functionality should be done using tanstack-query, just like in other places of our app, also axios should be used instead of fetch. This fetching functionality should be sepearate custom hook, along with TOKEN_CINFIG

Reply:
has been moved to a custom hook ser.